File: Swift/ExampleTests/common/AddressComponentsTyped.swift
public struct AddressComponentsTyped: Equatable, CustomStringConvertible {
    public let zip: String
    public let streetName: String
    public let city: String
    public let preDirection: String
    public let suffixDirection: String
    public let state: String
    public let suffixType: String

    public init(zip: String, streetName: String, city: String, preDirection: String, suffixDirection: String, state: String, suffixType: String) {
        self.zip = zip
        self.streetName = streetName
        self.city = city
        self.preDirection = preDirection
        self.suffixDirection = suffixDirection
        self.state = state
        self.suffixType = suffixType
    }

    public init(from s: AddressComponentsString) {
        self.zip = s.zip
        self.streetName = s.streetName
        self.city = s.city
        self.preDirection = s.preDirection
        self.suffixDirection = s.suffixDirection
        self.state = s.state
        self.suffixType = s.suffixType
    }

    public func toStringStruct() -> AddressComponentsString {
        return AddressComponentsString(
            zip: String(describing: zip),
            streetName: String(describing: streetName),
            city: String(describing: city),
            preDirection: String(describing: preDirection),
            suffixDirection: String(describing: suffixDirection),
            state: String(describing: state),
            suffixType: String(describing: suffixType)
        )
    }

    public static func toStringList(_ list: [AddressComponentsTyped]) -> [AddressComponentsString] {
        return list.map { $0.toStringStruct() }
    }

    public static func fromStringList(_ list: [AddressComponentsString]) -> [AddressComponentsTyped] {
        return list.map { AddressComponentsTyped(from: $0) }
    }

    public func toJSONValue() -> [String: Any] {
        return [
            "zip": zip,
            "streetName": streetName,
            "city": city,
            "preDirection": preDirection,
            "suffixDirection": suffixDirection,
            "state": state,
            "suffixType": suffixType,
        ]
    }

    public func toJSON() throws -> String {
        return try Json.write(toJSONValue())
    }

    public init(fromJSONValue m: [String: Any]) throws {
        self.zip = try Json.asString(Json.require(m, "zip"), "zip")
        self.streetName = try Json.asString(Json.require(m, "streetName"), "streetName")
        self.city = try Json.asString(Json.require(m, "city"), "city")
        self.preDirection = try Json.asString(Json.require(m, "preDirection"), "preDirection")
        self.suffixDirection = try Json.asString(Json.require(m, "suffixDirection"), "suffixDirection")
        self.state = try Json.asString(Json.require(m, "state"), "state")
        self.suffixType = try Json.asString(Json.require(m, "suffixType"), "suffixType")
    }

    public init(fromJSON text: String) throws {
        try self.init(fromJSONValue: Json.parseObject(text))
    }

    public static func toJSONList(_ list: [AddressComponentsTyped]) throws -> String {
        return try Json.write(list.map { $0.toJSONValue() })
    }

    public static func fromJSONList(_ text: String) throws -> [AddressComponentsTyped] {
        return try Json.parseArray(text).map {
            try AddressComponentsTyped(fromJSONValue: Json.asObject($0, "AddressComponentsTyped"))
        }
    }

    public var description: String {
        return "zip=\(zip), streetName=\(streetName), city=\(city), preDirection=\(preDirection), suffixDirection=\(suffixDirection), state=\(state), suffixType=\(suffixType)"
    }
}

File: Rust/ExampleTests/common/addresscomponents_typed.rs
#![allow(dead_code, unused_imports, unused_variables)]

use super::json;
use super::addresscomponents_string::AddressComponentsString;

#[derive(Debug, Clone, Default, PartialEq)]
pub struct AddressComponentsTyped {
    pub zip: String,
    pub streetname: String,
    pub city: String,
    pub predirection: String,
    pub suffixdirection: String,
    pub state: String,
    pub suffixtype: String,
}

impl AddressComponentsTyped {
    pub fn from_str_struct(s: &AddressComponentsString) -> Self {
        Self {
            zip: s.zip.clone(),
            streetname: s.streetname.clone(),
            city: s.city.clone(),
            predirection: s.predirection.clone(),
            suffixdirection: s.suffixdirection.clone(),
            state: s.state.clone(),
            suffixtype: s.suffixtype.clone(),
        }
    }

    pub fn to_str_struct(&self) -> AddressComponentsString {
        AddressComponentsString {
            zip: self.zip.clone(),
            streetname: self.streetname.clone(),
            city: self.city.clone(),
            predirection: self.predirection.clone(),
            suffixdirection: self.suffixdirection.clone(),
            state: self.state.clone(),
            suffixtype: self.suffixtype.clone(),
        }
    }

    pub fn to_string_list(list: &[AddressComponentsTyped]) -> Vec<AddressComponentsString> {
        list.iter().map(|t| t.to_str_struct()).collect()
    }

    pub fn from_string_list(list: &[AddressComponentsString]) -> Vec<AddressComponentsTyped> {
        list.iter().map(AddressComponentsTyped::from_str_struct).collect()
    }

    pub fn to_json_value(&self) -> json::Value {
        json::Value::Object(vec![
            ("zip".to_string(), json::Value::Str(self.zip.clone())),
            ("streetName".to_string(), json::Value::Str(self.streetname.clone())),
            ("city".to_string(), json::Value::Str(self.city.clone())),
            ("preDirection".to_string(), json::Value::Str(self.predirection.clone())),
            ("suffixDirection".to_string(), json::Value::Str(self.suffixdirection.clone())),
            ("state".to_string(), json::Value::Str(self.state.clone())),
            ("suffixType".to_string(), json::Value::Str(self.suffixtype.clone())),
        ])
    }

    pub fn to_json(&self) -> String {
        json::write(&self.to_json_value())
    }

    pub fn from_json_value(v: &json::Value) -> json::JsonResult<Self> {
        Ok(Self {
            zip: json::as_string(json::require(v, "zip")?, "zip")?,
            streetname: json::as_string(json::require(v, "streetName")?, "streetName")?,
            city: json::as_string(json::require(v, "city")?, "city")?,
            predirection: json::as_string(json::require(v, "preDirection")?, "preDirection")?,
            suffixdirection: json::as_string(json::require(v, "suffixDirection")?, "suffixDirection")?,
            state: json::as_string(json::require(v, "state")?, "state")?,
            suffixtype: json::as_string(json::require(v, "suffixType")?, "suffixType")?,
        })
    }

    pub fn from_json(text: &str) -> json::JsonResult<Self> {
        Self::from_json_value(&json::parse(text)?)
    }

    pub fn to_json_list(list: &[AddressComponentsTyped]) -> String {
        json::write(&json::Value::Array(
            list.iter().map(|item| item.to_json_value()).collect(),
        ))
    }

    pub fn from_json_list(text: &str) -> json::JsonResult<Vec<AddressComponentsTyped>> {
        let root = json::parse(text)?;
        let items = json::as_array(&root, "AddressComponentsTyped")?;
        let mut result = Vec::with_capacity(items.len());
        for e in items {
            result.push(Self::from_json_value(e)?);
        }
        Ok(result)
    }
}
